pkg/domain/infosync/info.go

```diff
@@ -129,23 +130,18 @@ type InfoSyncer struct {
 	infoCache             infoschemaMinTS
 }
 
-// ServerInfo is server static information.
-// It will not be updated when tidb-server running. So please only put static information in ServerInfo struct.
+// ServerInfo is server's basic information.
 type ServerInfo struct {
-	ServerVersionInfo
-	ID             string            `json:"ddl_id"`
-	IP             string            `json:"ip"`
-	Port           uint              `json:"listening_port"`
-	StatusPort     uint              `json:"status_port"`
-	Lease          string            `json:"lease"`
-	StartTimestamp int64             `json:"start_timestamp"`
-	Labels         map[string]string `json:"labels"`
-	// ServerID is a function, to always retrieve latest serverID from `Domain`,
-	// which will be changed on occasions such as connection to PD is restored after broken.
-	ServerIDGetter func() uint64 `json:"-"`
+	StaticServerInfo
+	DynamicServerInfo
```

Comment:
Seems DynamicServerInfo is also static because its value never changes for `ServerInfo`?

Reply:
It's updated in [`setDynamicServerInfo`](https://github.com/pingcap/tidb/blob/fbaad4bea8d4ebcf6733b4d46d3e9d6fe4445411/pkg/domain/infosync/info.go#L1557-L1565), which replaces `InfoSyncer.info` with a new `ServerInfo`.